### ISSUE TITLE
Бэкенд для модерации NE

### ISSUE THREAD

**Issue by madfriend** [OpenCorpora/opencorpora#101]
Нужно:
- в `lib/lib_ne.php`: подобие `become_syntax_moderator`, `finish_syntax_moderation` (, `set_syntax_annot_status` ?) 
- в таблице `books` поле для модератора NE
- `copy_entity` (или как-то так, смысл, наверное, понятен)
- проверить, что функции, получающие список NE и список комментариев, работают, если не указывать `user_id` 

Пока вроде все.

**Comment by grandsbor** [comment]
О, что есть

**Comment by madfriend** [comment]
Ага :)

**Comment by grandsbor** [comment]
cc @victorbocharov 
Наверно, надо заново продумать требования к бэкенду

**Comment by grandsbor** [comment]
cc @yourmary

**Comment by grandsbor** [comment]
Саш, тут всё перечислено, что надо?

**Comment by madfriend** [comment]
блин, прощу прощения, не там написал про нужные функции -- перепутал тикеты про модерацию ИС и NE. Копирую:

- (модератору) получить всю разметку абзаца (разбитую по юзерам)
- нужно поле с id модератора NE у книжек
- какой-то способ понять, ты ли модератор книжки
- ручки в апи: скопировать спан от одного юзера другому, скопировать упоминание (со всеми спанами)

**Comment by madfriend** [comment]
+ нужен небольшой рефакторинг get_books_with_ne: внутри она должна вызывать другую функцию (напр. get_all_books_with_ne; или этой добавить аргументы available и count) -- которая отдает список всех размечаемых книжек, а не только 10 доступных. Это нужно, чтобы модераторы могли просматривать полный список и заходить в книжки, которые они модерируют.
Функция get_all_ne_moderators тоже будет полезной (чтобы контролировать, кто видит полный список книжек)
cc @yourmary, @grandsbor

**Comment by yourmary** [comment]
сделала модераторскую разметку рядом с обычной - флаг в таблице ne_paragraphs, флаг у start_ne_annotation, проверки на этот флаг. копирование спана и упоминания - в аннотацию, а не в юзера.
вместо get_all_ne_moderators сделала get_users_by_permission.

**Comment by grandsbor** [comment]
Саш, тут что-то ещё осталось?

**Comment by madfriend** [comment]
вспомнил -- все пользователи могут вызывать ручку finishAnnotation (для абзаца, который начали размечать), а модератор еще должен иметь доступ к ручке reopenAnnotation (т.е. модерация еще не окончена). Правильно, Витя? cc @victorbocharov

**Comment by vbocharov** [comment]
Да. С точки зрения модератора нужно различать следующие состояния:
- текст никто не модерирует
- текст модерируется мной
- текст отмодерирован мной (такие тексты должны идти в экспорт)
Между этими состояниями нужно иметь возможность переключаться.

**Comment by madfriend** [comment]
именно текст, а не абзац?

**Comment by vbocharov** [comment]
Да, на уровне UI можно считать, что один текст - один модератор.

**Comment by grandsbor** [comment]
А точно нужно переключение, в состояние "текст никто не модерирует" из других состояний?

**Comment by grandsbor** [comment]
Можно это закрыть?

**Comment by madfriend** [comment]
Витя? Будет модерация 4 слоя, про факты?

**Comment by vbocharov** [comment]
Да, нужна возможность отказаться от модерации.

Про 4 слой и модерацию его, давайте обсудим на очной встрече. Мы же ещё не решили, куда идти со всем этим дальше.

**Comment by madfriend** [comment]
это какая-то новая возможность, которую мы не обсуждали. Я помню только, что ты хотел возможность на уровне книжки пометить, что модерация готова или не готова
